rust/worker/src/system/receiver.rs

```diff
@@ -65,15 +65,17 @@ impl ChromaError for ChannelError {
     }
 }
 
-#[derive(Error, Debug)]
-pub enum ChannelRequestError {
+#[derive(Error, Debug, PartialEq)]
+pub enum RequestError {
     #[error("Failed to send request")]
-    RequestError,
+    SendError,
     #[error("Failed to receive response")]
-    ResponseError,
+    ReceiveError,
+    #[error("Message handler panicked")]
+    HandlerPanic(Option<String>),
```

Comment:
could we make this  #[from(HandlerPanic(Box<dyn ChromaError>))] - that way the error is still there.

Reply:
sorry, not understanding, what do you mean "the error is still there"?

Reply:
Nevermind!